CMakeLists.txt

```diff
@@ -1,227 +1,182 @@
-cmake_minimum_required(VERSION 3.10)
+cmake_minimum_required(VERSION 3.20)
 project(
 	zenohc
 	VERSION 0.7.0.1
 	DESCRIPTION "The C bindings for Zenoh"
 	HOMEPAGE_URL "https://github.com/eclipse-zenoh/zenoh-c"
 	LANGUAGES C
 )
-set(CARGO_PROJECT_VERSION "${PROJECT_VERSION_MAJOR}.${PROJECT_VERSION_MINOR}.${PROJECT_VERSION_PATCH}")
-
-if(NOT PROJECT_VERSION_TWEAK)
-	set(CARGO_PROJECT_VERSION "${CARGO_PROJECT_VERSION}-dev")
-elseif(PROJECT_VERSION_TWEAK LESS 255)
-	set(CARGO_PROJECT_VERSION "${CARGO_PROJECT_VERSION}-rc.${PROJECT_VERSION_TWEAK}")
-endif()
 
-message(STATUS "CARGO_PROJECT_VERSION: ${CARGO_PROJECT_VERSION}")
-configure_file("${CMAKE_CURRENT_SOURCE_DIR}/Cargo.toml.in" "${CMAKE_CURRENT_SOURCE_DIR}/Cargo.toml" @ONLY)
+include(cmake/helpers.cmake)
+set_default_build_type(Release)
+enable_testing()	
 
-if(NOT CMAKE_BUILD_TYPE)
-	set(default_build_type "RelWithDebInfo")
-	message(STATUS "Setting build type to '${default_build_type}' as none was specified.")
-	set(CMAKE_BUILD_TYPE "${default_build_type}" CACHE
-		STRING "Choose the type of build." FORCE)
-
-	# Set the possible values of build type for cmake-gui
-	set_property(CACHE CMAKE_BUILD_TYPE PROPERTY STRINGS
-		"Debug" "Release" "MinSizeRel" "RelWithDebInfo")
-endif()
-
-
-if (NOT DEFINED CARGO_FLAGS)
-  set(CARGO_FLAGS "")
-endif()
+#
+# Build zenohc library from rust sources
```

Comment:
It's better to move zenohc's build script to separate *.cmake file! It is always a good approach to split large cmake projects to some kind of tree of smaller files containing atomic functionality.

Reply:
I thought about this. But I had a feeling that this separation would be too artificial.

The structure of CMake files follows directory structure of the project:
```
- library sources 
  | - examples
  | - tests
  | - install scripts
```

I think that it would be better to organize it like this:
- project root
  | - library sources
  | - examples
  | - tests
  | - install scripts
```

In this case the simple root CMakeLists.txt would include library sources as directory with it's own CMakeLists.txt which intended to build library only. But I don't want to make such big refactoring at this moment.

Reply:
Yes, I thought about refactoring like you showed, but I also thought that it'll be too much.